### PR TITLE
add validation for mnemonics on diff network

### PR DESCRIPTION
**Pull request by amiraabouhadid**
### Description

added mnemonic verification method to check if user provided mnemonics are for a user on the current network.

### Changes

- added a watcher for mnemonic that assigns public variable this.grid to gridclient fetched from entered mnemonics
- added a verification function to mnemonics rules to check if the returned grid has a twinId. If it doesnt then there isnt a user on the current network. 
- Profile manager now displays a `Couldn't find a user for provided mnemonic on ${config.network} network.` for invalid mnemonics that are bip39 compliant but do not belong to a user on the current network

### Related Issues

fixes issue #894 

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

https://github.com/threefoldtech/tfgrid-sdk-ts/assets/31689104/9f699879-5b8c-4e54-ac74-5f1da45db02b

### CODE REVIEW

File: packages/dashboard/src/components/TfChainConnector.vue [hunk 1]
```diff
@@ -277,11 +277,19 @@ export default class TfChainConnector extends Vue {
   public canLogin = typeof localStorage.getItem(key) === "string";
 
   public balance: any | null = null;
+  public twinID: any;
+  public grid: any;
+  @Watch("mnemonic")
+  async updateGrid() {
+    console.log(this.mnemonic);
```
Comment (0oM4R) [inline packages/dashboard/src/components/TfChainConnector.vue:284]: __Question__: is there any need for logging the mnemonic here ?

**Review by 0oM4R — COMMENTED** [review]
- if i paste my mnem it show the invalid message and the logs contain many errors, once i delete it and paste it again it works fine
- also on any change we have an error in the logs i believe we can handle it 
 
[Screencast from 09 أغس, 2023 EEST 12:16:56 م.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/722e695f-ad5b-4395-a3d9-c6c0c5371adb)

**Comment by amiraabouhadid** [comment]
<div style="position: relative; padding-bottom: 56.25%; height: 0;"><iframe src="https://www.loom.com/embed/1a7743feff954ae9bdcc0f55549aadde?sid=d4fa4b77-1fbc-4833-97bc-f8a0e7fa9afb" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>

**Comment by MohamedElmdary** [comment]
@amiraabouhadid I think we should disable `connect` button while validating using watcher as user might click it

File: packages/dashboard/src/components/TfChainConnector.vue [hunk 2]
```diff
@@ -277,11 +277,24 @@ export default class TfChainConnector extends Vue {
...
   public balance: any | null = null;
+  public twinID: any;
+  public grid: any;
+  @Watch("mnemonic")
+  async checkMnemonic() {
+    try {
+      this.grid = await getGrid(this.mnemonic);
+      this.grid.twinId;
+    } catch (err) {
+      this.mnemonicError = `Couldn't find a user for provided mnemonic on ${config.network} network.`;
```
Comment (0oM4R) [inline packages/dashboard/src/components/TfChainConnector.vue:288]: For better UX we can show 'Mnemonic is required' as an error message when the field is empty 
```suggestion
       if(!this.mnemonic) this.mnemonicError = "Mnemonic is required"
      else this.mnemonicError = `Couldn't find a user for provided mnemonic on ${config.network} network.`;
```

**Review by 0oM4R — CHANGES_REQUESTED** [review]

**Comment by Omarabdul3ziz** [comment]
- `connect` button still not disabled if no mnemonic is provided. only if i write something and remove it `Mnemonic is required` message appears and and `connect` get disabled.
  image

apart from that all looks good

**Review by Omarabdul3ziz — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**